### PR TITLE
fix: Ensure donations don't show up as sponsors

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What changes did you make? (Give an overview)

Siemens recently donated $1000 to ESLint through GitHub Sponsors, but it's showing up as a sponsor rather than a donation. 

This updates our script to filter out any one-time sponsorships from the sponsors list (these are displayed as individual donations rather than sponsorships).

I also fixed the sorting of donations so that the Open Collective and GitHub donations are sorted in chronological order across both data sets.

### CODE REVIEW

**Review by harish-sethuraman — DISMISSED** [review]
LGTM. Shall I send PR to update in other repos?

File: tools/fetch-sponsors.js [hunk 1]
```diff
@@ -387,7 +397,7 @@ async function fetchGitHubSponsors() {
 	const donations = openCollectiveDonations.concat(githubDonations);
 
 	// sort donations so most recent is first
-	donations.sort((a, b) => new Date(a) - new Date(b));
+	donations.sort((a, b) => Date.parse(b.date) - Date.parse(a.date));
```
Comment (fasttime) [inline tools/fetch-sponsors.js:400]: Good catch!

**Review by fasttime — APPROVED** [review]
LGTM, thanks!

**Comment by amareshsm** [comment]
Nice catch! I’m nearly done with the changes. Hopefully, this will resolve the issue in other repositories as well. You can track the progress here: https://github.com/eslint/eslint.org/issues/619.

**Pull request merged**